src/components/Main/Main.js

```diff
@@ -3,15 +3,29 @@ import Slider from "../Slider/Slider";
 import productCategoriesMock from "../../mocks/en-us/product-categories.json";
 import featuredBannersMock from "../../mocks/en-us/featured-banners.json";
 import featuredProductsMock from "../../mocks/en-us/featured-products.json";
+import ProductListPage from "../ProductListPage/ProductListPage";
 import FeaturedProducts from "../FeaturedProducts/FeaturedProducts";
 
-const Main = () => {
+const Main = ({showHomePage, showAllProducts, onViewChange} ) => {
 
+   
   return (
     <main>
-        <Slider items={featuredBannersMock.results} />
-        <Carousel items={productCategoriesMock.results} />
-        <FeaturedProducts items={featuredProductsMock.results} />
+      {showHomePage && (
+        <>
+          <h1>Home Page</h1>
+          <Slider items={featuredBannersMock.results} />
+          <Carousel items={productCategoriesMock.results} />
+          <FeaturedProducts items={featuredProductsMock.results} handleViewChange={onViewChange}/>
+        </>
+      )}
+
+      {showAllProducts && (
+        <>
+          <button onClick={() => onViewChange(true, false)}>Home Page</button>
+          <ProductListPage items={featuredProductsMock.results} />
+        </>
+      )}
```

Comment:
I don't see value on adding this component, you can move this code to the `App` component, to make it more simpler.